src/event/mod.rs

```diff
@@ -119,7 +119,7 @@ impl LogEvent {
         self.fields.contains_key(key)
     }
 
-    pub fn insert<K, V>(&mut self, key: K, value: V)
+    pub fn insert<K, V>(&mut self, key: K, value: V) -> Option<Value>
```

Comment:
That's something that I considered to add when implementing nested fields! I decided against back then on the grounds that no component used it at the moment.

I like that the resulting API becomes more and more aligned with the APIs of the collections from the Rust's standard library, I think it should reduce cognitive overhead from using them.

Reply:
It's my secret goal to make `Event` just a serde_json, bytes, or a simdjson object, so that sounds about right. :)